packages/tool-cache/src/tool-cache.ts

```diff
@@ -254,6 +255,35 @@ export async function extractTar(
   return dest
 }
 
+/**
+ * Extract a xar compatible archive
+ *
+ * @param file     path to the archive
+ * @param dest     destination directory. Optional.
+ * @param flags    flags for the xar. Optional.
+ * @returns        path to the destination directory
+ */
+export async function extractXar(
+  file: string,
+  dest?: string,
+  flags: string = '-x'
+): Promise<string> {
+  ok(IS_MAC, 'extractXar() not supported on current OS')
+  ok(file, 'parameter "file" is required')
+
+  dest = dest || (await _createExtractFolder(dest))
```

Comment:
_createExtractFolder(dest) is a no-op if the directory already exists. The other extraction functions actually call it if a destination is passed in, so it will create the destination if it does not already exist.

This is a helpful for users, and we should do it here as well.

`dest = await _createExtractFolder(dest)`

Reply:
We should also add a unit test for this functionality 
`extract .xar to a directory that does not exist`